### ISSUE TITLE
Style/ArgumentsForwarding incorrectly fixes a case where an argument is used elsewhere in the method

### ISSUE THREAD

**Issue by davidenglishmusic** [rubocop/rubocop#12089]
The auto-correct for the cop can break code by replacing the arguments when one of them is referenced in another part of the method.

--------

## Expected behavior
The following should not elicit a correction:

```
def method_missing(m, *args, **kwargs, &block)
  if @template.respond_to?(m)
    @template.send(m, *args, **kwargs, &block)
  else
    super
  end
end
```

## Actual behavior
It corrects to the following which throws an undefined variable error:

```
def method_missing(...)
  if @template.respond_to?(m)
    @template.send(...)
  else
    super
  end
end
```

## Steps to reproduce the problem

Auto-correct the code listed under the expected behavior heading.

## RuboCop version

```
1.55.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.1) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.23.1
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rspec 2.23.0
```

**Comment by mattbrictson** [comment]
I noticed a similar bug.

Original:

```ruby
def post(*args, &block)
  future_on(executor, *args, &block)
```

Erroneously corrected by rubocop to:

```ruby
def post(...)
  future_on(...)
```

Expected behavior:

```ruby
def post(...)
  future_on(executor, ...)
```

**Comment by tmikoss** [comment]
Adding another similar example. Produced by `rubocop -a`, version `1.55.1`. `association_name` is used in the fucntion body.

<img width="519" alt="Screenshot 2023-08-07 at 10 31 07" src="https://github.com/rubocop/rubocop/assets/222126/d502549d-eeb8-44d3-9b44-bcec339fc2d7">

**Comment by koic** [comment]
FYI, I've opened #12101 to fix due to many reports about this issue. Thank you.

**Comment by chulkilee** [comment]
[1.56](https://github.com/rubocop/rubocop/releases/tag/v1.56.0) included the fix (#12103) but it did not fix the issue.

Source

```ruby
class Foo
  def bar(*args, &block)
    hello(*args, &block)
    world(*args)
  end
end
```

Script

```ruby
bin/rubocop --version
1.56.0

bin/rubocop -a test.rb
Inspecting 1 file
C

Offenses:

test.rb:2:11: C: [Corrected] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
  def bar(*args, &block)
          ^^^^^^^^^^^^^
test.rb:3:11: C: [Corrected] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
    hello(*args, &block)
          ^^^^^^^^^^^^^
```

Output

```ruby
class Foo
  def bar(...)
    hello(...)
    world(*args)
  end
end
```

Can we reopen this issue? Or let me know if opening a new issue is preferred.

**Comment by owst** [comment]
@chulkilee apologies that I've still not got the cop quite right - without looking at the code, your issue looks like a slightly different scenario/cause to me, so I think it should be a new issue. I'll try and take a look over the weekend.

**Comment by chulkilee** [comment]
@owst Got it - created #12117
